### DISCUSSION TITLE
does envoy custom dashboard behave differently than other dashboards?

### DISCUSSION THREAD

**Discussion by jmazzitelli** [kiali/kiali#4381] [Q&A] [answered]
Does Kiali treat the Envoy custom dashboard "special" or differently than the other custom dashboards?

Because I do not see the Envoy custom dashboard on applications, but I do see it on workloads.
However, the Go and Kiali custom dashboards show up on both applications and workloads.

For example, deploy Kiali and bookinfo and look at bookinfo's "details" and istio-system's "kiali" applications and workloads (replace "localhost:20001" with your kiali host:port)

1 Application: details -> http://localhost:20001/kiali/console/namespaces/bookinfo/applications/details?duration=60&refresh=15000

2 Workload: details -> http://localhost:20001/kiali/console/namespaces/bookinfo/workloads/details-v1?duration=60&refresh=15000

3 Application: kiali -> http://localhost:20001/kiali/console/namespaces/istio-system/applications/kiali?duration=60&refresh=15000

4 Workload: kiali -> http://localhost:20001/kiali/console/namespaces/istio-system/workloads/kiali?duration=60&refresh=15000

The 'Envoy" custom dashboard tab only shows up on 2 but not 1.
However, the "Go" and "Kiali" custom dashboard tabs show up on both 3 and 4.

Here's where the envoy tab is missing:
![image](https://user-images.githubusercontent.com/2029470/134149203-c2754cdb-b296-4c70-a5de-56f5e592bc54.png)

**Comment by lucasponce** [comment, marked as answer]
Yes, I remember that we considered part of the core infrastructure and the Envoy dashboard is special (it's filtered or treated differently), but it shares some logic with the CustomDashboard.

Probably it would make sense to move it as part of the Istio ones, but we didn't see the tradeoff to do it.
